### ISSUE TITLE
Phase 4: Update Prediction Model for Two-Tier AFCF Methodology

### ISSUE THREAD

**Issue by reggiechan74** [reggiechan74/issuer-credit-analysis#45]
## Problem

The distribution cut prediction model (v2.1) was trained on **total AFCF** (including all investing activities), but Phase 3 now calculates **two-tier AFCF** with sustainable AFCF as the primary metric (v1.0.14, Issue #40).

**Current Flow:**
1. Phase 3 calculates `afcf` = **sustainable AFCF** (recurring CFI only)
2. Phase 3 calculates `afcf_self_funding_ratio` = sustainable AFCF / total obligations
3. Phase 4 enrichment extracts `self_funding_ratio` from `afcf_coverage.afcf_self_funding_ratio`
4. Prediction model uses `self_funding_ratio` as a feature (5th most predictive)

**Issue:** Model was trained on total AFCF but now receives sustainable AFCF, causing:
- Feature distribution shift (sustainable AFCF < total AFCF when non-recurring items exist)
- Potential prediction accuracy degradation
- Risk misclassification (model calibrated on different metric)

## Evidence

**Phase 3 Output Structure (lines 1214-1225 in `calculate_credit_metrics.py`):**
```python
result = {
    # PRIMARY METRIC: Sustainable AFCF (recurring CFI only)
    'afcf': round(afcf_sustainable, 0),
    'afcf_sustainable': round(afcf_sustainable, 0),
    'net_cfi_sustainable': round(net_cfi_sustainable, 0),

    # COMPARISON: Total AFCF (all CFI components)
    'afcf_total': round(afcf_total, 0),
    'net_cfi_total': round(net_cfi_total, 0),
    ...
}
```

**Phase 4 Enrichment (line 420 in `enrich_phase4_data.py`):**
```python
afcf_coverage = phase3.get('afcf_coverage', {})
fundamentals['self_funding_ratio'] = afcf_coverage.get('afcf_self_funding_ratio', 0)
```

**Model Features (15 selected):**
- Rank 4: `self_funding_ratio` (coefficient: +0.871)
- Heavily weighted feature in prediction

## Impact Assessment

**Severity:** HIGH
- `self_funding_ratio` is the #1 most predictive feature (coef: +0.871)
- Artis REIT example:
  - Sustainable AFCF: -$33M
  - Total AFCF (if property dispositions +$50M): -$33M + $50M = +$17M
  - Self-funding ratio shift: -0.34x → +0.17x (51 basis point change)
  - This could shift probability from 5.4% to different range

**Credit Analysis Impact:**
- Risk levels may be miscalibrated
- Distribution cut probability predictions may be inaccurate
- Model performance metrics (F1: 0.813) based on total AFCF training

## Recommended Solution

**Option 1: Retrain Model with Sustainable AFCF (Recommended)**
- Update training dataset to use sustainable AFCF methodology
- Retrain model on all 24 REIT observations
- Validate new model performance (target: F1 ≥ 0.80)
- Update model version to v2.2
- Benefits:
  - Aligns with AFFO/ACFO sustainability principles
  - More conservative metric for credit analysis
  - Consistent with Phase 3 methodology

**Option 2: Use Total AFCF for Prediction Model**
- Update Phase 4 enrichment to use `afcf_total` for model features
- Calculate separate self-funding ratios:
  - `self_funding_ratio_sustainable`: For credit analysis (primary)
  - `self_funding_ratio_total`: For prediction model (model input)
- Benefits:
  - No model retraining required
  - Preserves existing model calibration
  - Quick fix

**Option 3: Feature Engineering**
- Add both ratios as features: `self_funding_ratio_sustainable` + `self_funding_ratio_total`
- Retrain model with expanded feature set (16 features instead of 15)
- Let model learn from both metrics
- Benefits:
  - Most comprehensive approach
  - Captures both sustainability and total cash generation

## Acceptance Criteria

- [ ] Confirm which AFCF version was used in original training (v2.1)
- [ ] Choose solution approach (Option 1, 2, or 3)
- [ ] Update code as needed:
  - [ ] Phase 3 calculations (if Option 2/3)
  - [ ] Phase 4 enrichment feature extraction
  - [ ] Model retraining (if Option 1/3)
- [ ] Test with 3 REITs (RioCan, Artis, Dream Industrial):
  - [ ] Verify predictions are reasonable
  - [ ] Compare predictions before/after fix
  - [ ] Validate self-funding ratio values
- [ ] Update documentation:
  - [ ] CLAUDE.md (note model version and AFCF methodology)
  - [ ] Model metadata (if retrained)
  - [ ] Phase 4 agent instructions (if methodology changes)
- [ ] Regenerate enriched data and reports for all 3 test REITs

## Related Issues

- Issue #40: Two-Tier AFCF Methodology Implementation (v1.0.14)
- Issue #39: OpenBB Platform Integration (market/macro/distribution data)
- Issue #38: Distribution Cut Prediction Model v2.0

## Files Affected

- `scripts/calculate_credit_metrics.py` (Phase 3 AFCF calculation)
- `scripts/enrich_phase4_data.py` (Phase 4 feature extraction)
- `models/distribution_cut_logistic_regression.pkl` (if retraining)
- Training scripts (if retraining)
